src/main/java/core/basesyntax/db/StorageImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageImpl implements Storage {
+    private Map<String, Integer> fruitMap;
+
+    public StorageImpl() {
+        fruitMap = new HashMap<>();
+    }
+
+    @Override
+    public void add(String fruit, int quantity) {
+        fruitMap.put(fruit, quantity);
```

Comment:
The `add` method should ensure that the quantity is updated rather than replaced. If the fruit already exists in the map, its quantity should be increased by the provided `quantity`.